Examples/src/main/kotlin/com/mapbox/vision/examples/UsbVideoSourceActivityKt.kt

```diff
@@ -0,0 +1,229 @@
+            }
+            backgroundHandler.post {
+                this@UsbVideoSourceActivityKt.videoSourceListener = videoSourceListener
+                // Init and register USBMonitor.
+                usbMonitor = USBMonitor(this@UsbVideoSourceActivityKt, onDeviceConnectListener)
+                usbMonitor?.register()
+            }
+        }
+
+        override fun detach() {
+            backgroundHandler.post {
+                uvcCamera?.stopPreview()
+
+                usbMonitor?.unregister()
+                usbMonitor?.destroy()
+                videoSourceListener = null
+            }
+
+            backgroundHandlerThread.quitSafely()
+        }
+    }
+
+    private var usbMonitor: USBMonitor? = null
+    private var uvcCamera: UVCCamera? = null
+
+    // VisionEventsListener handles events from Vision SDK on background thread.
+    private val visionEventsListener = object : VisionEventsListener {}
+
+    override fun onPermissionsGranted() {
+        startVisionManager()
+    }
+
+    override fun initViews() {
+        setContentView(R.layout.activity_main)
+    }
+
+    override fun onStart() {
+        super.onStart()
+        startVisionManager()
+    }
+
+    override fun onStop() {
+        super.onStop()
+        stopVisionManager()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        vision_view.onResume()
+    }
+
+    override fun onPause() {
+        super.onPause()
+        vision_view.onPause()
+    }
+
+    private fun startVisionManager() {
+        if (allPermissionsGranted() && !visionManagerWasInit) {
+            VisionManager.create(usbVideoSource)
+            VisionManager.setModelPerformanceConfig(
+                ModelPerformanceConfig.Merged(
+                    ModelPerformance.On(ModelPerformanceMode.FIXED, ModelPerformanceRate.HIGH)
+                )
+            )
+            vision_view.setVisionManager(VisionManager)
+            VisionManager.visionEventsListener = visionEventsListener
+            VisionManager.start()
+
+            visionManagerWasInit = true
+        }
+    }
+
+    private fun stopVisionManager() {
+        if (visionManagerWasInit) {
+            VisionManager.stop()
+            VisionManager.destroy()
+
+            visionManagerWasInit = false
+        }
+    }
+
+    private val onDeviceConnectListener: USBMonitor.OnDeviceConnectListener =
+        object : USBMonitor.OnDeviceConnectListener {
+            override fun onAttach(device: UsbDevice?) {
+                toast("Usb camera attached")
+                usbMonitor?.requestPermission(device!!)
+            }
+
+            override fun onConnect(
+                device: UsbDevice?,
+                ctrlBlock: USBMonitor.UsbControlBlock?,
+                createNew: Boolean
+            ) {
+                toast("Usb camera connected")
+                backgroundHandler.post {
+                    synchronized(this@UsbVideoSourceActivityKt) {
```

Comment:
Why do we need `synchronized` here?

Reply:
I've followed UVCCamera impementation of listener at the moment, will review it once again now

Reply:
So it seems that callbacks can be fired from multiple threads so access to usb* classes are synchronized. Added even one more synchronized block after check)